### DISCUSSION TITLE
Should we use TitleBar or Page? Im confused

### DISCUSSION THREAD

**Discussion by artyrcheek** [Shopify/polaris-react#8648] [Q&A]
I'm seeing the use of the <Page> component recommended, but I thought the TitleBar was the way to go?
TitleBar seems to be better with the ContextualSaveBar

**Comment by chloerice** [comment]
Hey @artyrcheek 👋🏽 This is a great question. The `Page` component should be used as a layout only component for embedded apps. So you'd wrap your UI in `Page`, rendering `TitleBar` for the actions, title, etc instead of setting those props on `Page`. Check out [this section](https://shopify.dev/docs/apps/tools/app-bridge/actions/titlebar#using-titlebar-with-polaris) of the TitleBar docs for more complete example code, but here's some sudo code for the difference between how Page headers are built internally vs in embedded apps:

```diff
- <Page title="Title" primaryAction={{content: 'Save'}}>
+ <Page>
+  <TitleBar title="Title" primaryAction={{content: 'Save'}} />
</Page>
```
